src/middlewares/__tests__/createInsightsMiddleware.ts

```diff
@@ -86,25 +86,6 @@ describe('insights', () => {
   });
 
   describe('initialize', () => {
-    it('initialize insightsClient', () => {
```

Comment:
This test is redundant. We do not need _hasCredentials anywhere in search-insights or InstantSearch.js. (The test may have been added to see the possibility of avoiding duplicated init calls, but it's taken care in another PR.)